### PR TITLE
Switch to kconfig

### PR DESCRIPTION
**Pull request by mbolivar**
This series:

- starts with a cleanup of the Kconfig file
- adds new options to replicate functionality handled via CMake CONF_xxx macros in the Zephyr CMakeLists.txt
- mandates that every supported platform provide an mcuboot_config/mcuboot_config.h (only Mynewt was providing this previously)
- migrates each CONF_xxx over to its Kconfig equivalent by progressively extending Zephyr's mcuboot_config.h
- finishes up with a little extra cleanup

Fixes: #186 

Other than bootutil_log.h (which is itself a platform abstraction layer), this has the side effect of removing all platform-specific ifdeffery in bootutil behind the MCUBOOT_xxx configuration variables defined in mcuboot_config.h.

### CODE REVIEW

**Comment by mbolivar** [comment]
@SebastianBoe FYI

**Comment by mbolivar** [comment]
I ran the simulator test and "go run run-tests.go" from zephyr/samples; no regressions.

Some Mynewt testing would be much appreciated!

**Comment by utzig** [comment]
LGTM. No regressions found on Mynewt. Please fix the simulator with signature tests and we can merge afterwards.

File: docs/PORTING.md [hunk 1]
```diff
@@ -46,6 +50,16 @@ After running the management functions of the bootloader, `boot_go` returns
...
+## Configuration file
+
+You must provide a file, mcuboot_config/mcuboot_config.h. This is
+included by several files in the "library" portion of MCUboot; it
+provides preprocessor definitions that configure the library's
+build.
+
+See the file samples/mcuboot_config/mcuboot_config.template.h for a
+starting point and more information.
+
```
[the remark this answers is not in the document]
Comment (mbolivar) [reply docs/PORTING.md]: The first comment in the template file does point at Zephyr's and Mynewt's files, but I'm fine to add that here too.

**Comment by mbolivar** [comment]
> Please fix the simulator with signature tests and we can merge afterwards.

Interesting, I didn't know "cargo test" took a --features flag that could enable extra testing! That's not documented in the README.rst, so I'll add another patch on that makes it clear "cargo test" doesn't cover everything.

**Comment by mbolivar** [comment]
@utzig @carlescufi I've updated the series to address your comments.

I ran all the simulator --features flags manually as well, so hopefully CI passes this time. I had no idea just how much the simulator "borrowed" files from the Zephyr port until I wrote this series!

**Pull request merged**